plugins/postcss-cascade-layers/src/index.ts

```diff
@@ -14,13 +14,47 @@ function postcssCascadeLayers(opts) {
 					atRule.params = `anon${layerCount}`;
 				}
 
+				let hasNestedLayers = false;
+				let hasUnlayeredStyles = false;
+
+				// check for where a layer has nested layers AND styles outside of those layers
+				atRule.each((node) => {
+					if (node.type == "atrule") {
+						hasNestedLayers = true;
+					} else if (node.type == "rule") {
+						hasUnlayeredStyles = true;
+					}
+				});
+
+				if (hasNestedLayers && hasUnlayeredStyles) {
+					let cloned = [];
+					//create final layer
+					atRule.append({
```

Comment:
Can you point to the specification were it says that implicit inner layers come after other inner layers?

https://www.w3.org/TR/css-cascade-5/#layer-ordering

I am unsure if adding it after is always correct.